sdk/storage/azure-storage-blob-v2/src/main/java/com/azure/v2/storage/blob/ContainerClient.java

```diff
@@ -0,0 +1,1255 @@
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     */
+    @Metadata(properties = { MetadataProperties.GENERATED })
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public void create(String containerName) {
+        this.serviceClient.create(containerName);
+    }
+
+    /**
+     * returns all user-defined metadata and system properties for the specified container. The data returned does not
+     * include the container's list of blobs.
+     * 
+     * @param containerName The name of the container.
+     * @param timeout The timeout parameter is expressed in seconds. For more information, see &lt;a
+     * href="https://docs.microsoft.com/en-us/rest/api/storageservices/fileservices/setting-timeouts-for-blob-service-operations"&gt;Setting
+     * Timeouts for Blob Service Operations.&lt;/a&gt;.
+     * @param leaseId If specified, the operation only succeeds if the resource's lease is active and matches this ID.
+     * @param requestOptions The options to configure the HTTP request before HTTP client sends it.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws StorageErrorException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return the response.
+     */
+    @Metadata(properties = { MetadataProperties.GENERATED })
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Response<Void> getPropertiesWithResponse(String containerName, Integer timeout, String leaseId,
+        RequestOptions requestOptions) {
+        return this.serviceClient.getPropertiesWithResponse(containerName, timeout, leaseId, requestOptions);
+    }
+
+    /**
+     * returns all user-defined metadata and system properties for the specified container. The data returned does not
+     * include the container's list of blobs.
+     * 
+     * @param containerName The name of the container.
+     * @param timeout The timeout parameter is expressed in seconds. For more information, see &lt;a
+     * href="https://docs.microsoft.com/en-us/rest/api/storageservices/fileservices/setting-timeouts-for-blob-service-operations"&gt;Setting
+     * Timeouts for Blob Service Operations.&lt;/a&gt;.
+     * @param leaseId If specified, the operation only succeeds if the resource's lease is active and matches this ID.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws StorageErrorException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     */
+    @Metadata(properties = { MetadataProperties.GENERATED })
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public void getProperties(String containerName, Integer timeout, String leaseId) {
```

Comment:
We'll want to make a possible change with how codegen is working to only provide maximal overloads for `WithResponse` APIs.